server/world/world.go

```diff
@@ -1449,6 +1566,60 @@ func (w *World) tickEntities(tick int64) {
 	w.entitiesToTick = w.entitiesToTick[:0]
 }
 
+// SetRaining toggles raining depending on the raining argument.
+func (w *World) SetRaining(raining bool, x time.Duration) {
+	w.mu.Lock()
+	w.setRaining(raining, x)
+	w.mu.Unlock()
+}
+
+func (w *World) setRaining(raining bool, x time.Duration) {
+	level := 0
+	if raining {
+		level = 1
+	}
+	w.setRainLevel(level)
+
+	for _, v := range w.allViewers() {
+		v.ViewRain(raining)
+	}
+
+	if raining {
+		w.setRainTime(int(x.Seconds() * 20))
+	} else {
+		w.setRainTime(rand.Intn(168000) + 12000)
+	}
+}
+
+// SetThunder toggles raining depending on the thundering argument.
+func (w *World) SetThunder(thundering bool, x time.Duration) {
+	w.mu.Lock()
+	w.setThunder(thundering, x)
+	w.mu.Unlock()
+}
+
+func (w *World) setThunder(thundering bool, x time.Duration) {
+	if thundering && !w.isRaining() {
+		w.setRaining(true, time.Duration(rand.Intn(600)+600))
+	}
+
+	level := 0
+	if thundering {
+		level = 1
+	}
+	w.setThunderLevel(level)
+
+	for _, v := range w.allViewers() {
+		v.ViewThunder(thundering)
+	}
+
+	if thundering {
+		w.setThunderTime(int(x.Seconds() * 20))
+	} else {
+		w.setThunderTime(rand.Intn(168000) + 12000)
```

Comment:
This duration might be wrong?

Reply:
It is not. `setThunderTime` is in ticks, so this is valid.
The ones that require a `time.Duration` are in seconds, so these have lower values.